backend/src/controllers/middlewaresControllers/createAuthMiddleware/register.js

```diff
@@ -0,0 +1,84 @@
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+const Joi = require('joi');
+
+const mongoose = require('mongoose');
+
+const checkAndCorrectURL = require('./checkAndCorrectURL');
+const sendMail = require('./sendMail');
+
+const { loadSettings } = require('@/middlewares/settings');
+const { useAppSettings } = require('@/settings');
+
+const register = async (req, res, { userModel }) => {
+  const UserPasswordModel = mongoose.model(userModel + 'Password');
+  const UserModel = mongoose.model(userModel);
+  const { name, email, password, country } = req.body;
+
+  // validate
+  const objectSchema = Joi.object({
+    email: Joi.string()
+      .email({ tlds: { allow: true } })
+      .required(),
+    name: Joi.string().required(),
+    country: Joi.string().required(),
+    password: Joi.string().required(),
+  });
+
+  const { error, value } = objectSchema.validate({ name, email, password, country });
+  if (error) {
+    return res.status(409).json({
+      success: false,
+      result: null,
+      error: error,
+      message: 'Invalid/Missing credentials.',
+      errorMessage: error.message,
+    });
+  }
+
+  const user = await UserModel.findOne({ email: email, removed: false });
+  if (user)
+    return res.status(409).json({
+      success: false,
+      result: null,
+      message: 'An account with this email already exists.',
+    });
+
+  //  authUser if your has correct password
+  const salt = await bcrypt.genSalt(10);
+  console.log(salt)
+  const hashedPassword = await bcrypt.hash(password, salt);
```

Comment:
As with the comment above, we could reuse existing hashing logic from [here](https://github.com/idurar/idurar-erp-crm/blob/master/backend/src/models/coreModels/AdminPassword.js). Maybe @salahlalami can confirm